IMPLNOTES.md

```diff
@@ -0,0 +1,21 @@
+# Implementation notes
+
+A core goal of this decorators proposal is to be implementation-friendly, both to tooling and to end-to-end JS implementations. This file contains some rough notes on how implementations might be organized. If you work on a JavaScript implementation of any kind, and you see any issues with this proposal or approach, please [file an issue](https://github.com/tc39/proposal-decorators/issues/new).
+
+Decorators are specified as if they execute entirely at runtime. However, the limited expressiveness of user-defined decorators means that it's always predicable, if you've parsed all of the imported decorators, which built-in decorators the user-defined decorator ends up expanding out into. This property can be taken advantage of by implementations in varied ways.
+
+## Transpiler implementations
+
+Nicolò Ribaudo has written [notes](https://hackmd.io/44ErLPn8Qi6FshyoTcrXcA?view#) about a possible implementation strategy for Babel. To summarize:
+- An additional file format, `.decorators.json`, would summarize what built-in decorators each exported decorator boils down to. This would be made available to the importing module, when transpiling.
+- A runtime representation of decorators also has to exist, including name-mangled functions which execute these and can be imported by other modules.
+
+If possible, it would be great to agree on common formats for these two items among tooling. If consensus can be reached among participating tooling implementations, the intention would be that IMPLNOTES.md will eventually contains a description of this shared format.
```

Comment:
```suggestion
If possible, it would be great to agree on common formats for these two items among tooling. If consensus can be reached among participating tooling implementations, the intention would be that IMPLNOTES.md will eventually contain a description of this shared format.
```